### ISSUE TITLE
ImportError: Master does not have 'termios'

### ISSUE THREAD

**Issue by moreati** [dw/mitogen#121]
Commit a9c6c139b407d28ad4522611648b5b27442faa54 (importer has priority over system packages when whitelisting is enabled) breaks a large portion of the test suite for me. e.g.

```
python tests/nested_test.py 
No handlers could be found for logger "mitogen"
E
======================================================================
ERROR: test_nested (__main__.NestedTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/nested_test.py", line 13, in test_nested
    context = self.router.local(via=context, name='local%d' % x)
  File "/home/alex/src/mitogen/mitogen/master.py", line 644, in local
    return self.connect('local', **kwargs)
  File "/home/alex/src/mitogen/mitogen/parent.py", line 452, in connect
    return self.proxy_connect(via, method_name, name=name, **kwargs)
  File "/home/alex/src/mitogen/mitogen/parent.py", line 461, in proxy_connect
    name, context_id, method_name, kwargs
  File "/home/alex/src/mitogen/mitogen/master.py", line 612, in call
    return self.call_async(fn, *args, **kwargs).get().unpickle()
  File "/home/alex/src/mitogen/mitogen/core.py", line 314, in unpickle
    raise obj
mitogen.core.CallError: exceptions.ImportError: Master does not have 'termios'
  File "<stdin>", line 1383, in _dispatch_calls
  File "<stdin>", line 1370, in _dispatch_one
  File "<stdin>", line 588, in load_module
  File "master:/home/alex/src/mitogen/mitogen/parent.py", line 38, in <module>
    import termios
  File "<stdin>", line 567, in load_module


----------------------------------------------------------------------
Ran 1 test in 0.109s

FAILED (errors=1)
```

I identified the commit with bisect. I presume it's something to do with using a virtualenv, but I haven't confirmed that.

**Comment by dw** [comment]
I knew overriding the whitelist semantics was a bad idea anyway, but didn't realize it was quite so broken :( Sorry about this, I was in a rush..

Thinking through the obvious fixes, this is kinda can-of-wormy and I'm a little tired :) Will sort it first thing tomorrow.

One fix might simply be to add yet another kind of list passed through to the children, but this feels like tacking more mess on top of mess (the need for those lists at all is IMHO already a design disaster)

**Comment by dw** [comment]
This is a fun one :)  Simply removing that code unveils another issue: "ImportError: Cannot re-init internal module __main__", which looks to be due to the replacement of ``__import__`` with ``imp.find_module()`` as part of the fix for #113.

Having a second try at this later today

**Comment by dw** [comment]
Hi Alex,

I'm going to make an explicit push to get better at testing now this has users, and get away from "lol just a hobby project" mentality. I've defined a new issue label, NeedsRegressionTest, please feel free to apply it liberally to any open and closed bugs. Also if there are similar labels for tracking test quality and whatnot, let me know and I'm happy to add more.
